src/dialects/builtin/types.rs

```diff
@@ -220,11 +219,11 @@ impl UnitType {
 impl Printable for UnitType {
     fn fmt(
         &self,
-        ctx: &Context,
+        _ctx: &Context,
         _state: &printable::State,
         f: &mut core::fmt::Formatter<'_>,
     ) -> core::fmt::Result {
-        write!(f, "{}", Self::get_type_id_static().disp(ctx),)
+        write!(f, "()")
```

Comment:
Do we need the `()`, given that the name is printed, and that `()` carries no extra info. If you think we should have it, then the parser for this also need to be updated. Right now, it doesn't parse anything and just returns the type.

Reply:
I originally introduced the `()` because we used to print the typename followed by a space.
For now I removed the space from the `Typed` printable.